File: AlbumLibrary/src/main/java/com/album/AlbumImageLoader.java
package com.album;

import android.content.Context;
import android.support.annotation.NonNull;
import android.widget.ImageView;

import com.album.model.AlbumModel;
import com.album.model.FinderModel;
import com.album.ui.annotation.FrescoType;

/**
 * by y on 15/08/2017.
 */

public interface AlbumImageLoader {

    void displayAlbum(@NonNull ImageView view, int width, int height, @NonNull AlbumModel albumModel);

    void displayAlbumThumbnails(@NonNull ImageView view, @NonNull FinderModel finderModel);

    void displayPreview(@NonNull ImageView view, @NonNull AlbumModel albumModel);

    ImageView frescoView(@NonNull Context context, @FrescoType int type);
}

File: core/src/main/java/com/album/core/action/AlbumImageLoader.kt
package com.album.core.action

import android.view.View
import android.widget.FrameLayout
import com.album.scan.ScanEntity

/**
 * @author y
 * @create 2019/2/27
 */
interface AlbumImageLoader {
    /**
     * 首页图片加载
     */
    fun displayAlbum(width: Int, height: Int, albumEntity: ScanEntity, container: FrameLayout): View

    /**
     * 目录图片加载
     */
    fun displayAlbumThumbnails(finderEntity: ScanEntity, container: FrameLayout): View

    /**
     * 预览页图片加载
     */
    fun displayAlbumPreview(albumEntity: ScanEntity, container: FrameLayout): View
}